app/reducers/book_reducer.rb

```diff
@@ -0,0 +1,9 @@
+class BookReducer < Rack::Reducer
+    def initialize()
+        super(Book.all, 
+            -> (author:) { where('lower(author) like ?', "%#{author.downcase}%") },
+            -> (genre:) {where('lower(genre) like ?', "%#{genre.downcase}%")}, 
+            -> (title:) {where('lower(title) like ?', "%#{title.downcase}%")} 
+        )
+    end
+end
```

Comment:
Add newline at end of file

Reply:
a line break?

Reply:
All files should include an empty line at end of the file

Reply:
ok!